### ISSUE TITLE
Add capability to parse additional FFmpeg output

### ISSUE THREAD

**Issue by st-h** [kokorin/Jaffree#38]
First off, thanks for creating this project. I came here from ffmpeg-cli-wrapper and finally found most of the flexibility I need when using FFmpeg. However, I came along one thing I am missing:

The ability to retrieve and parse FFmpeg command line output (usually errout). For instance when using the [loudnorm](http://k.ylo.ph/2016/04/04/loudnorm.html) filter, which essentially is a two pass job that requires to retrieve and parse the parameters from the first run and use them for the second run.

I was thinking about adding a capability to add a custom OutputParser to `FFmpegResultReader`:

```
public FFmpegResult read(InputStream stdOut) {
        //just read stdOut fully
        BufferedReader reader = new BufferedReader(new InputStreamReader(stdOut));
        String errorMessage = null;

        String line;
        FFmpegResult result = null;

        try {
            while ((line = reader.readLine()) != null) {

                if (outputParser != null) {
                    outputParser.parse(line);
                }
...
```

OutputParser is just an interface:
```
public interface OutputParser {
    public void parse(String line);
}
```

in the end this will be invoked like:
```
LoudnormParser loudnormParser = new LoudnormParser()

FFmpegResult result = FFmpeg.atPath()
                .setOutputParser(loudnormParser)
                .addInput(UrlInput.fromPath(Paths.get("/Users/steve/Desktop/3-part Harm 1b_keyCmin_56bpm.wav")))
                .addArguments("-af", "loudnorm=I=-16:TP=-1.5:LRA=11:print_format=json")
                .addOutput(new NullOutput())
                .execute();

loudnormParser.getInputThreshold();
```
where LoundnormParser is an implementation of OutputParser.
When the command has been executed, one can obtain parsed values from the OutputParser implementation.

I will be happy to submit a PR if this sounds like a reasonable thing to add to you. And please let me know in case you come along better names 😸

**Comment by kokorin** [comment]
Thanks for you interest in Jaffree. My heart melts with such worm words.

This functionality seems reasonable. I'm not sure if every line should be passed to custom parser or only those lines, which are not progress or result.

Also it's possible to parse/read both stderr and stdout. Have to check to which one ffmpeg writes extra data.

If you need this feature ASAP, it's better to fork Jaffree and implement it yourself. (with PR if possible).
I will try to check it closer this weekend.

Denis.

**Comment by st-h** [comment]
@kokorin I just like to check before putting together a PR 😃  For now I was able to achieve something similar by subclassing a few of the provided classes (I think I am good for now without an imminent release). Downside is that I had to pretty much duplicate the `FFmpegResultReader` because of all the parse methods that are private. 
I was also thinking about providing some kind of abstract Reader class which provides all the parse methods. Then both the `FFmpegResultReader` and any custom parser could extend that class and use the existing parse methods.

> I'm not sure if every line should be passed to custom parser or only those lines, which are not progress or result.

I think the difficult thing might be to find out if the current line really is just something that should only be handled by the progress or result parser. Maybe passing those lines to another parser won't be too much overhead compared to what FFmpeg actually is doing? However, I haven't really found a way to deal with the FFmpeg output that I really would be happy with.

> Also it's possible to parse/read both stderr and stdout. Have to check to which one ffmpeg writes extra data.

From my observations, the stdout is pretty quiet and I read somewhere that FFmpeg usually outputs to stderr. But I honestly do not have a lot of experience with FFmpeg itself.
